HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,36 @@
+def extraArgs(neighbors,distance):
+	"""If a node is in neighbors, and it hasn't been visited, return it. Else return"""
+	def updateNode(n):
+		if n[0] in neighbors and n[1] > distance+1:
+			# if n[3] == False:
+			# 	accum2.add(1)
+			if n[3] == False:
+				#accum.add(1)
+				return (n[0],distance+1,n[2],True)
+			else:
+				return (n[0],distance+1,n[2],n[3])
+		else:
+			return n
+	return updateNode
+
+def f(neighbors):
+	return lambda y:y[3]==False and y[0] in neighbors
+
```

Comment:
Sorry for my lack of comments. y[3] tells me if the node has been visited or not. y[0] is name of node.
This function returns true if the node has not been visited and is in neighbors.